federatedscope/core/fed_runner.py

```diff
@@ -89,7 +91,18 @@ def _setup_for_standalone(self):
                 self.cfg.federate.sample_client_num = 1
                 self.cfg.freeze()
 
+        unseen_clients_id = []
+        if self.cfg.federate.unseen_clients_rate > 0:
+            unseen_clients_id = np.random.choice(
+                np.arange(1, self.cfg.federate.client_num + 1),
+                size=max(
+                    1,
+                    int(self.cfg.federate.unseen_clients_rate *
+                        self.cfg.federate.client_num)),
+                replace=False).tolist()
+
         self.server = self._setup_server()
+        self.server.unseen_clients_id = unseen_clients_id
```

Comment:
IMO, we have better not modify the attributes of server/client in fed_runner

Reply:
moved it into the setup_server() as sub action of init